### PR TITLE
Initial Starbound Generic Config

### PR DESCRIPTION
**Pull request by IceOfWraith**
This is Windows only at the moment due to the working directory only able to point at one location. Windows is under win64 and Linux is under linux directories.

### CODE REVIEW

**Comment by IceOfWraith** [comment]
This game has a Query Port, but it doesn't seem to use it. I don't see any documentation lately about it either. In game connects like normal.

Please feel free to test at your convenience!

File: starboundconfig.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    "ParamFieldName": "rconServerPassword",
+    "IncludeInCommandLine": false,
+    "DefaultValue": "",
+    "EnumValues": {},
+    "Hidden": true
+  },
+  {
+    "DisplayName": "RCON Enabled",
+    "Category": "Starbound - Settings",
+    "Description": "RCON Password.",
```
Comment (Greelan) [inline starboundconfig.json:87]: Doesn't strictly matter since it is hidden, but the description is incorrect

**Review by Greelan — CHANGES_REQUESTED** [review]
Have done a desktop review as it seems I also need to own this game to install the server

File: starboundconfig.json (hunk 1 above)
Comment (IceOfWraith) [reply starboundconfig.json]: Addressed as well as changed the category for easy spotting in the config manifest.

File: starbound.kvp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+App.UniversalSleepSteamQueryPortRef=gameServerPort
+App.MaxUsers=8
+App.UseRandomAdminPassword=True
+App.RemoteAdminPassword=Password123
+App.AdminMethod=SourceRCON
+App.AdminLoginTransform=None
+App.RCONConnectDelaySeconds=5
+App.RCONConnectRetrySeconds=5
+App.TelnetLoginFormat={0}
+App.UpdateSources=[{"UpdateStageName": "SteamCMD Download","UpdateSourcePlatform": "All", "UpdateSource": "SteamCMD", "UpdateSourceData": "211820"},{"UpdateStageName":"Create Logs Directory","UpdateSourcePlatform":"All","UpdateSource":"CreateDirectory","UpdateSourceArgs":"./starbound/211820/storage","OverwriteExistingFiles":false},{"UpdateStageName":"Config File Download","UpdateSourcePlatform":"All","UpdateSource":"FetchURL","UpdateSourceData":"https://github.com/CubeCoders/AMPTemplates/main/starbound_server.config","UpdateSourceTarget":"{{$FullBaseDir}}storage/","OverwriteExistingFiles":false},{"UpdateStageName":"Create Symlink","UpdateSourcePlatform":"Linux","UpdateSource":"Executable","UpdateSourceData":"/bin/ln","UpdateSourceArgs":"-s {{$FullBaseDir}}linux {{$FullBaseDir}}win64"}]
```
[the remark this answers is not in the document]
Comment (IceOfWraith) [reply starbound.kvp:54]: Game devs, what are you gonna do? ¯\_(ツ)_/¯

File: starbound.kvp [hunk 3]
```diff
@@ -0,0 +1,83 @@
+Meta.DisplayName=Starbound
+Meta.Description=Starbound Dedicated Server
+Meta.OS=Windows, Linux
+Meta.Arch=x86_64
+Meta.Author=IceOfWraith
+Meta.URL=https://store.steampowered.com/app/211820/Starbound/
+Meta.DisplayImageSource=steam:211820
+Meta.EndpointURIFormat=
```
[the remark this answers is not in the document]
Comment (IceOfWraith) [reply starbound.kvp:8]: As discussed, the Steam Query port for this server doesn't seem to function, so I excluded this to avoid confusion.

**Review by Greelan — APPROVED** [review]

**Review by Greelan — APPROVED** [review]

**Pull request merged**